tests/unit/Service/SQS.service.test.js

```diff
@@ -129,14 +139,48 @@ describe('Service/SQS', () => {
       });
     });
 
-    it('throws an error if publish fails', async () => {
+    it(`catches the error if publish fails with failureMode === ${SQS_PUBLISH_FAILURE_MODES.CATCH}`, async () => {
+      const service = getService({
+        sendMessage: new Error('SQS is down!'),
+      }, false);
+
+      const promise = service.publish(TEST_QUEUE, { test: 1 }, null, SQS_PUBLISH_FAILURE_MODES.CATCH);
+
+      await expect(promise).resolves.toEqual(null);
```

Comment:
Since we've got a spy on `Logger.error`, could we also assert that it gets called with the expected error?

Reply:
As discussed, this is probably not worth testing as it starts to tie into the implementation details.